custom/abdm/auth.py

```diff
@@ -0,0 +1,19 @@
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+from custom.abdm.models import ABDMUser
+
+
+class ABDMUserAuthentication(TokenAuthentication):
+
+    def authenticate(self, request):
+        access_token = request.META.get('HTTP_AUTHORIZATION', "")
+        token = access_token.split(" ")[-1]
+        if not token:
+            return None
+        try:
+            user = ABDMUser.objects.get(access_token=token)
```

Comment:
If you make this model comply with the rest framework key structure then you could just do this:
```
class ABDMUserAuthentication(TokenAuthentication):
    model = ABDMUser
```

A rest framework token has a 'key' and 'user' fields (user is a FK to the Django user).

I'm also unclear why you need a custom model for the token and can't either use the existing HQApiKey or else the DRF Token model.

Reply:
Intention is to restrict token's access to ABDM API only as it is accessed by a separate app. Will it possible to implement this limitation without a custom auth logic?

Reply:
I remember that if those entries related to DRF's token were not added in the `settings.py` file, it resulted in the authentication system not getting triggered.

Reply:
ok